addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2760,7 +2773,11 @@ var SnippetsMenu = Widget.extend({
 
         // Prepare the functions
         const functions = {};
-        if (noCheck) {
+        // In translate mode, it is only possible to modify text content but not
+        // the structure of the snippets. For this reason, the "Editable area"
+        // are only the text zones and they should not be used inside functions
+        // such as "is", "closest" and "all".
+        if (noCheck || this.options.enableTranslation) {
```

Comment:
I also had to add this :point_up: in order for the code to work correctly

Reply:
Mmh, I understand your explanation but in which case are those "is", etc functions called during translate mode? :thinking: And why is it a problem in 16.0 and above only?

Reply:
> In which case are those "is", etc functions called during translate mode?

When you click on a snippet on the "Invisible elements" panel to display it. Indeed, it comes from `start()` of the `SnippetEditor` -> `_initializeOptions()` -> `if (!val.selector.is(this.$target))`.

> And why is it a problem in 16.0 and above only?

It seems that it is only since [1] that `is()` can depend on `getEditableArea()`. Note that  I realized that this fix was needed tanks to this `is()` function. Indeed, without it, when clicking on an invisible element, `_initializeOptions()` will not work correctly as `this.styles` will remain empty. Due to that, `toggleTargetVisibility()` will also fail to toggle the visibility of the snippet as `this.styles` was not initialized correctly.

`all()` and `closest()` (that were influenced by `getEditableArea()` in 15.0) are also called in translate mode (for example `all()` is called in `_disableUndroppableSnippets()` when entering edit mode and `closest()`  is called in `_createSnippetEditor()` when clicking on the snippet on the "Invisible Elements" panel). However, I did not realize it when working on the version 15.0 as it does not lead to "visual" bugs.

[1]: https://github.com/odoo/odoo/commit/86e3bb5e24c2d125ec4ec84ac788601c63182d0f